src/promo/entity/promo.entity.ts

```diff
@@ -0,0 +1,39 @@
+import type { IPromoCode } from 'promo/promo.interface';
+import { PromocodeState } from 'promo/promocode.enum';
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'promocode' })
+export class PromocodeEntity implements IPromoCode {
+  @PrimaryGeneratedColumn()
+    id: string;
+
+  @Column()
+    name: string;
+
+  @Column()
+    createdAt: string;
+
+  @Column()
+    percent: number;
+
+  @Column()
+    currentState: PromocodeState;
+
+  @Column()
+    isOneTime: boolean;
+
+  @Column()
+    usedDate: string;
```

Comment:
as far as i remember we set this field if the code is one time used. that's why string | null